src/error.rs

```diff
@@ -0,0 +1,87 @@
+//! Implements Error type. It implements the `std::error::Error` for errors returned by ODBC and
+//! allows for an error handling which his idomatic to rust
+
+use super::raw;
+use std::fmt::{Display, Formatter};
+use std;
+
+/// ODBC Diagnostic Record
+#[derive(Debug)]
+pub struct DiagRec {
+    pub state: [raw::SQLCHAR; 6],
+    pub native_error_pointer: raw::SQLINTEGER,
+    pub message: String,
+}
+
+impl DiagRec {
+    /// Reads first diagnostic record from handle
```

Comment:
Thanks for you feedback! Here is my rational, for why I handled it this way:
1. About panicking on SQL_NO_DATA:
Maybe the name "create" is a bit to general. I did not intend to write a general wrapper around DiagRec, but only one that servers the specific purpose of picking up an error right after it happened. The function is only called if the postcondition of a previous function call guarantees that there is at least one diagnostic record (usually after an SQL Error). In an intermediate state I had already entertained the idea of returning an option (to avoid panicking on SQL_NO_DATA), but that just meant that I had to unwrap that option higher up in the stack or present the user with an option which is never `none`.
2. About returning only the first record instead of collecting all:
I could be wrong here, but currently I'm not sure if there can be more than one diagnostic record related to the error, the way we currently use the API. Indead the odbc specification allows us only to assume that there is at least one record, but I think a driver is most likely to produce a multiple diagnostic records in cases there the successful execution of a statement produces a number of warnings. Long story short: I will feel more comfortable implementing this, if I have the means to test it. It is highly likely I will revisit the subject when I know how to actually produce multiple diagnostic records. Do you know any specific cases in which such a thing would occur?
On a more general note: How do you feel about warnings, hm, maybe I should open an issue for this...